tests/appsec/test_processor.py

```diff
@@ -18,13 +19,35 @@
 RULES_MISSING_PATH = os.path.join(ROOT_DIR, "nonexistent")
 
 
+class Config(object):
+    def __init__(self):
+        self.is_header_tracing_configured = False
+
+
 def _enable_appsec(tracer):
     tracer._appsec_enabled = True
     # Hack: need to pass an argument to configure so that the processors are recreated
     tracer.configure(api_version="v0.4")
     return tracer
 
 
+def test_transform_headers():
+    assert _transform_headers(
+        {
+            "hello": "world",
+            "Foo": "bar1",
+            "foo": "bar2",
+            "fOO": "bar3",
+            "BAR": "baz",
+            "COOKIE": "secret",
+        }
+    ) == {
+        "hello": "world",
+        "foo": ["bar1", "bar2", "bar3"],
+        "     bar": "baz",
```

Comment:
Does this still need to be updated?
```suggestion
        "bar": "baz",
```